Source/CarthageKit/GitHub.swift

```diff
@@ -148,7 +148,13 @@ extension Client {
 			Client.userAgent = gitHubUserAgent()
 		}
 
-		let urlSession = URLSession.proxiedSession
+        var urlSession: URLSession
+        
+        if server.url.host?.contains("github.com") == true {
+            urlSession = URLSession.proxiedSessionWithDelegate
+        } else {
+            urlSession = URLSession.proxiedSession
+        }
```

Comment:
Any idea why this is needed? The delegate implementation below seems like it should be correct—even for enterprise repositories.

My preference would be to fix the delegate if possible instead of introducing a second proxy.

Reply:
Unfortunately I haven't been able to figure that out yet. I tried implementing all the delegate methods to see if the session was getting caught up in one of them and none of them seemed to get hit. The other strange thing is that a `carthage update` command works normally while the `carthage build` does not. I have not dug into the differences in the implementations yet to figure out why that may be.

Reply:
I have some time to look into this more today and I think I was wrong in thinking that the issue was with GitHub Enterprise. I think the issue is with downloading binaries and might affect GitHub dependencies as well. Still in the process of tracking down the issue.

Reply:
I've tracked the way that Carthage "hangs" into ReactiveSwift in FoundationExtensions.swift. Given this code:

```swift
extension Reactive where Base: URLSession {
	/// Returns a SignalProducer which performs the work associated with an
	/// `NSURLSession`
	///
	/// - parameters:
	///   - request: A request that will be performed when the producer is
	///              started
	///
	/// - returns: A producer that will execute the given request once for each
	///            invocation of `start()`.
	///
	/// - note: This method will not send an error event in the case of a server
	///         side error (i.e. when a response with status code other than
	///         200...299 is received).
	public func data(with request: URLRequest) -> SignalProducer<(Data, URLResponse), AnyError> {
		return SignalProducer { [base = self.base] observer, lifetime in
			print("here in data with request: \(request)")
			
			let task = base.dataTask(with: request) { data, response, error in
				print("here in task completion handler: \(error)")
				if let data = data, let response = response {
					observer.send(value: (data, response))
					observer.sendCompleted()
				} else {
					observer.send(error: AnyError(error ?? defaultSessionError))
				}
			}

			lifetime.observeEnded(task.cancel)
			task.resume()
		}
	}
}
```

The first print statement gets hit 4 times (which makes sense because the concurrent downloads are limited to 4) but the second print statement never gets hit.

I'm out of ideas and energy at this point.

Reply:
I'm able to download a binary from GitHub with `master`. 🤷🏻‍♂️